### PR TITLE
Action list

### PR DESCRIPTION
**Pull request by alschmiedt**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
Keyboard navigation did not work in readonly mode. 

### Proposed Changes
If we are in read only mode and keyboard accessibility mode is turned on then we will allow a certain subset of actions that do not modify the workspace, but allow the user to move around the workspace.

### Test Coverage

Tested on:

### CODE REVIEW

File: core/keyboard_nav/ast_node.js [hunk 1]
```diff
@@ -288,7 +288,7 @@ Blockly.ASTNode.prototype.findPreviousEditableField_ = function(location,
   var previousField = null;
   var startIdx = opt_last ? fieldRow.length - 1 : fieldIdx - 1;
   for (var i = startIdx, field; field = fieldRow[i]; i--) {
-    if (field.isCurrentlyEditable()) {
```
Comment (RoboErikG) [inline core/keyboard_nav/ast_node.js]: Why this change?

File: core/blockly.js [hunk 2]
```diff
@@ -205,6 +205,13 @@ Blockly.svgResize = function(workspace) {
 // are multiple workspaces and non-main workspaces are able to accept input.
 Blockly.onKeyDown_ = function(e) {
   var mainWorkspace = Blockly.mainWorkspace;
+
+  if (mainWorkspace.options.readOnly &&
```
Comment (RoboErikG) [inline core/blockly.js]: I think you're conflicting a bit with the if statement immediate below. I'd actually move this below, take out the readOnly check in the other if so it reads like:

```
if (Blockly.utils.isTargetInput(e) || (...rendered && ... isVisible)) {
  ...
}
if (mainWorkspace.options.readOnly) {
  if (Blockly.keyboardAccessibilityMode) {
    Blockly.navigation.onKeyPress(e);
  }
  return; // Doesn't return a value
}
```

File: core/keyboard_nav/navigation.js [hunk 3]
```diff
@@ -837,14 +837,19 @@ Blockly.navigation.onKeyPress = function(e) {
...
   var curNode = Blockly.navigation.cursor_.getCurNode();
+  var readOnly = Blockly.getMainWorkspace().options.readOnly;
   var actionHandled = false;
 
   if (action) {
-    if (curNode && curNode.getType() === Blockly.ASTNode.types.FIELD) {
-      actionHandled = curNode.getLocation().onBlocklyAction(action);
-    }
-    if (!actionHandled) {
+    if (readOnly && Blockly.navigation.READONLY_ACTION_LIST.indexOf(action) > -1) {
```
Comment (RoboErikG) [inline core/keyboard_nav/navigation.js]: If you swap these two it becomes
if (!readonly) {
  ...
} else if (Blockly.navigation.READONLY_ACTION_LIST.indexOf(action) > -1) {
  ...
}

File: core/keyboard_nav/ast_node.js (hunk 1 above)
Comment (alschmiedt) [reply core/keyboard_nav/ast_node.js]: Because isCurrentlyEditable will always be false for fields when in readOnly mode. So the user wouldn't be able to navigate to the fields.

File: core/blockly.js (hunk 2 above)
Comment (alschmiedt) [reply core/blockly.js]: fixed

File: core/keyboard_nav/navigation.js (hunk 3 above)
Comment (alschmiedt) [reply core/keyboard_nav/navigation.js]: fixed

File: core/blockly.js [hunk 4]
```diff
@@ -206,20 +206,23 @@ Blockly.svgResize = function(workspace) {
...
     // Ignore keypresses on rendered workspaces that have been explicitly
     // hidden.
     return;
   }
+
+  if (mainWorkspace.options.readOnly) {
+    // When in read only mode handle key actions for keyboard navigation.
+    if (Blockly.keyboardAccessibilityMode &&
+      Blockly.navigation.onKeyPress(e)) {
+      return true;
```
Comment (RoboErikG) [inline core/blockly.js]: This method has no return. You can just check keyboardAccessibilityMode and then call onKeyPress without the return true.
